code/mob/living/critter/mimic.dm

```diff
@@ -0,0 +1,129 @@
+/mob/living/critter/mimic
+	name = "mechanical toolbox"
+	desc = null
+	icon_state = null
+	is_npc = TRUE
+	ai_type = /datum/aiHolder/mimic
+	can_lie = FALSE
+	butcherable = FALSE
+	health_brute = 20
+	health_burn = 20
+	health_brute_vuln = 0.75
+	health_burn_vuln = 1.25
+	hand_count = 1
+	add_abilities = list(/datum/targetable/critter/mimic, /datum/targetable/critter/tackle, /datum/targetable/critter/sting/mimic)
+	ai_retaliate_patience = 0
+	ai_retaliate_persistence = RETALIATE_UNTIL_INCAP
+	ai_retaliates = TRUE
+
+	dir_locked = TRUE //most items don't have dirstates, so don't let us change one
+	var/mutable_appearance/disguise
+	var/icon/face_image
+	var/icon/face_displace_image
+	var/is_hiding = FALSE
+
+	New()
+		..()
+		src.face_image = icon('icons/misc/critter.dmi',"mimicface")
+		var/toolboxType =pick(25;/obj/item/storage/toolbox/mechanical, 25;/obj/item/storage/toolbox/emergency, 25;/obj/item/storage/toolbox/electrical, 24;/obj/item/storage/toolbox/artistic, 1;/obj/item/storage/toolbox/memetic)
+		var/obj/item/storage/toolbox/startDisguise = new toolboxType(src)
+
+
+		src.disguise_as(startDisguise)
+		startDisguise.set_loc(null) //don't just have a random toolbox inside mimics
+		qdel(startDisguise)
+
+	update_icon()
+		. = ..()
+		//src.appearance = src.disguise
+		if(!is_hiding)
+			src.add_filter("mimic_face", 101, layering_filter(icon = src.face_image,  blend_mode = BLEND_INSET_OVERLAY))
+
+	setup_healths()
+		add_hh_flesh(src.health_brute, src.health_brute_vuln)
+		add_hh_flesh_burn(src.health_burn, src.health_burn_vuln)
+
+	setup_hands()
+		..()
+		var/datum/handHolder/HH = hands[1]
+		HH.limb = new /datum/limb/mouth
+		HH.icon = 'icons/mob/critter_ui.dmi'
+		HH.icon_state = "mouth"
+		HH.name = "mouth"
+		HH.limb_name = "teeth"
+		HH.can_hold_items = 0
+
+	death(gibbed, do_drop_equipment)
+		if(!gibbed)
+			src.visible_message("[src] explodes in a shower of meat!")
+			return src.gib()
```

Comment:
Should prooobably always call parent here?

Reply:
Can't, it's an infinite loop. `gib()` calls `death()`